hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -1422,6 +1423,14 @@ public List<BlockGroup> getPendingDeletionKeys(final int keyCount)
             //  4. Further optimization: Skip all snapshotted keys altogether
             //  e.g. by prefixing all unreclaimable keys, then calling seek
 
+            // If the bucket is a snapshot bucket then we should not process
+            // entries related to the snapshot from the deletedTable.
+            boolean isSnapshotBucket =
+                OMClientRequestUtils.isSnapshotBucket(this, info);
+            if (isSnapshotBucket) {
+              break;
+            }
```

Comment:
We do that object ID check in SnapshotDeletingService. The flow goes like this, correct me if I'm wrong

1. `getPendingDeletionKeys()` is called from KDT and it doesn't remove those keys from `deletedTable` that belong to the snapshot bucket.
2. #4280 Removes the keys within the snapshot scope from the `deletedTable` in active DB. Snapshot DB will still have those keys within the snapshot scope.
3. `SnapshotDeletingService` does the splitting of `RepeatedOmKeyInfo` and does an object ID match with the previous snapshot.

Since the object ID check is being done in SDS do we also need to do it here?

Reply:
Oh I see. I misunderstood earlier.

However, we are not still not checking whether the key can be **actually** reclaimed or not in active FS. For example, in the following case `key2` is not reclaimed (but can be):

1. Taken `snap1` on `/vol1/buck1/`.
2. Create new key `/vol1/buck1/key2`
3. Delete key `/vol1/buck1/key2` just created. Thus `/vol1/buck1/key2` is appended to active `deletedTable`.
4. With the current logic, `/vol1/buck1/key2` won't be reclaimed by KDT.

Reply:
Thanks for pointing that out. Makes sense. I will update the PR